Pretty board

emashliles opened the pull request:
@gemcfadyen @jsuchy

emashliles committed:
Print board with additional lines and spaces.

emashliles committed:
Remove one magic number from board printing.

emashliles committed:
Evaluate returns no winner.

jsuchy commented on src/com/TTT/BoardPrinter.java:
```diff
@@ -10,11 +10,19 @@ public BoardPrinter(PrintStream out) {
 
     public void printBoard(Board board) {
-        for(int i = 0; i < board.size(); i++)
+        int rowSize = 3;
```
Lots of magic numbers in here, with just the one extracted out. I'd suggest having a think about how you may be able to simplify this code. I would be very hesitant to change this code due to its complexity and differing levels of abstraction. Could use some effective refactoring to make it clear what's going on (I don't want to think about the maths in the main algorithm for printing the board. It should read clearly, with those details hidden away in methods that express the intent of each calculation / operation.

emashliles committed:
Move some setup to before.

emashliles committed:
Catch specific exception when parsing an int.

emashliles committed:
Fix formatting.

emashliles committed:
Remove checked exception.

jsuchy commented:
Nice small / focused commits here.

emashliles committed:
Begin moving validate logic to Game.

emashliles committed:
Add more validation to Game.

emashliles committed:
Move all validation logic to Game.

emashliles committed:
Rename test.

emashliles committed:
Remove more magic numbers.

emashliles commented:
I believe I have addressed most of the comments. I think I would like to move some of the validation logic around a bit still, bit I also need to get it finished.

emashliles committed:
Introduce Row object to make win checking easier to manage.
